packages/itwin/tree-widget/src/tree-widget-react/components/trees/common/components/Tree.tsx

```diff
@@ -156,39 +122,85 @@ function TreeBaseImpl({
     },
   });
   useIModelChangeListener({
-    imodel,
+    imodel: props.imodel,
     action: useCallback(() => {
       TreeWidget.logger.logTrace(`${LOGGING_NAMESPACE}.${treeName}`, `iModel data changed`);
       imodelChanged.raiseEvent();
     }, [imodelChanged, treeName]),
   });
 
+  if (treeProps.rootErrorRendererProps !== undefined) {
+    return <StrataKitRootErrorRenderer {...treeProps.rootErrorRendererProps} />;
+  }
+
+  return (
+    <TreeBase
+      {...props}
+      isReloading={isReloading}
+      treeRendererProps={treeProps.treeRendererProps}
+      getNode={getNode}
+      imodelChanged={imodelChanged}
+      getSchemaContext={getSchemaContext}
+      currentHierarchyLevelSizeLimit={currentHierarchyLevelSizeLimit}
+    />
+  );
+}
+
+type TreeBaseProps = {
+  currentHierarchyLevelSizeLimit: number;
+  getNode: (nodeId: string) => PresentationHierarchyNode | undefined;
+  imodelChanged: BeEvent<() => void>;
+  treeRendererProps?: TreeRendererProps;
+} & Omit<TreeProps, "selectionStorage" | "treeName" | "getHierarchyDefinition"> &
+  Pick<ReturnType<typeof useTree>, "getNode" | "isReloading">;
+
+/** @internal */
+export function TreeBase({ getSchemaContext, getFilteredPaths, onReload, treeRendererProps, ...props }: TreeBaseProps) {
+  if (treeRendererProps === undefined) {
+    return <SkeletonTree />;
+  }
+
+  return (
+    <SchemaMetadataContextProvider imodel={props.imodel} schemaContextProvider={getSchemaContext}>
+      <TreeBaseImpl {...props} treeRendererProps={treeRendererProps} />
+    </SchemaMetadataContextProvider>
+  );
+}
+
+function TreeBaseImpl({
+  imodel,
+  emptyTreeContent,
+  currentHierarchyLevelSizeLimit,
+  selectionPredicate,
+  selectionMode,
+  imodelChanged,
+  treeRenderer,
+  highlight,
+  treeRendererProps,
+  isReloading,
+  ...rest
```

Comment:
Does not look like other props from rest are used.
```suggestion
  getNode
```